src/backend/marsha/core/tests/test_views_lti_respond.py

```diff
@@ -56,50 +62,184 @@ def test_views_lti_respond(self):
+    )
+    @mock.patch("oauthlib.oauth1.rfc5849.generate_timestamp", return_value="1616018589")
+    def test_views_lti_respond_signature(self, mock_ts, mock_nonce):
+        """Validate signed data of the response returned by the view for an instructor request."""
+        lti_select_form_data, _ = generate_passport_and_signed_lti_parameters(
+            url="http://lti-consumer.site/lti/respond/",
+            lti_parameters={
+                "roles": "instructor",
+                "content_item_return_url": "http://lti-consumer.site/",
+                "context_id": "unknown",
+            },
+            passport_attributes={
+                "oauth_consumer_key": "W1PWSNDUL7T7YGMCOWZH",
+                "shared_secret": "passport_secret",
+            },
+        )
+        jwt_token = AccessToken()
+        jwt_token.payload["lti_select_form_data"] = lti_select_form_data
+        response = self.client.post(
+            "/lti/respond/",
+            {
+                "jwt": str(jwt_token),
+                "content_items": "some content items",
+            },
+        )
+        context = response.context_data
+
+        form_data = context.get("form_data")
+        self.assertEqual(
+            "xaB0rmss7hEJCNBVPwPPsXzeUWI=", form_data.get("oauth_signature")
         )
+        self.assertEqual(mock_ts.return_value, form_data.get("oauth_timestamp"))
+        self.assertEqual(mock_nonce.return_value, form_data.get("oauth_nonce"))
 
-    def test_views_lti_respond_verification_wrong_signature(self):
-        """Wrong signature should raise a 403 error."""
-        lti_parameters, _ = generate_passport_and_signed_lti_parameters(
+    def test_views_lti_respond_wrong_jwt(self):
+        """Wrong JWT should raise a 403 error."""
+        lti_select_form_data, _ = generate_passport_and_signed_lti_parameters(
             url="http://testserver/lti/respond/",
             lti_parameters={
                 "roles": random.choice(["instructor", "administrator"]),
-                "content_item_return_url": "http://return.url/",
+                "content_item_return_url": "http://lti-consumer.site/",
+                "context_id": "unknown",
+            },
+        )
+        jwt_token = AccessToken()
+        jwt_token.payload["lti_select_form_data"] = lti_select_form_data
+        response = self.client.post(
+            "/lti/respond/",
+            {
+                "jwt": str(jwt_token) + "a",
                 "content_items": "some content items",
+            },
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def test_views_lti_respond_no_jwt(self):
+        """Missing JWT should raise a 403 error."""
+        response = self.client.post(
+            "/lti/respond/",
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def test_views_lti_respond_no_content_items(self):
+        """Missing content_items post data should raise a 403 error."""
```

Comment:
Should be a 400 and not a 403 no ?

Reply:
As the main reason for this to append is a token alteration, I thought it could raise a 403.
But I'm ok with a 400 too :+1:

Reply:
Why a token modification ? this property is directly retrieved in the request body, not in the token:

```
{"content_items": self.request.POST.get("content_items")}
```

Reply:
Indeed, you are right, sorry.
Error 400 is set now.